### DISCUSSION TITLE
Sleep-Timer option: stop at chapter end

### DISCUSSION THREAD

**Discussion by xcy7e** [PaulWoitaschek/Voice#1613] [Ideas]
When using the sleep timer, would be nice to have a checkbox to set that the stop will only occur on the closest or next chapter end.

- [x] Stop at chapter end

 This would avoid interrupted storylines overnight

**Comment by tomaThomas** [comment]
I implemented sleep at end of the next chapter option (https://github.com/tomaThomas/Voice) because I want to use it myself

A setting could be added that would start this option after a normal sleep timer.
However, the implementation is a bit hacky (it periodically checks if the next chapter is reached, similarly to the normal sleep timer), but as I don't know Kotlin and the android media player very well, this is all I could do in reasonable time.

Also, I'm not sure what is best for the UI, I just added another floating action button, but this may need explanation...
![voice](https://user-images.githubusercontent.com/17884154/208250838-e2ea99f8-cd1d-4dd2-b007-c96c92476c93.png)

**Comment by marksz** [comment]
I like this idea!  Do you stop at the first chapter end, or 20 minutes, which ever comes first?
Alternatively it could play a minimum of 20 minutes and then stop at the next chapter end.  
If the playback speed is not 1, I'm assuming this button would work the same as the existing timer that counts clock time, not book position time, so at 1.5 speed 20 minutes could cover a 30 minute chapter.

**Reply by tomaThomas** [reply]
For now it is completely separate, so always the end of the first chapter if you press the left button.
I thought about doing your second suggestion, starting a normal sleep timer of 20 minutes (30 min chapter at 1.5 speed), and afterwards start the end of next chapter sleep timer. The UI could be changed to a toggle instead of two buttons.

I opened a pull request, I'll wait what Paul Woitaschek thinks of it

**Comment by sabi0** [comment]
Duplicates #1323

**Reply by eode** [reply]
Maybe folks can upvote #1323 if they haven't already.
